src/main/java/cd/go/plugin/config/yaml/JsonnetConfigParser.java

```diff
@@ -0,0 +1,76 @@
+package cd.go.plugin.config.yaml;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+
+import cd.go.plugin.config.yaml.transforms.RootTransform;
+
+public class JsonnetConfigParser extends YamlConfigParser {
+    public JsonnetConfigParser() {
+        super(new RootTransform());
+    }
+
+    @Override
+    public JsonConfigCollection parseFiles(File baseDir, String[] files) {
+        JsonConfigCollection collection = new JsonConfigCollection();
+        for (String file : files) {
+            try {
+                File filePath = new File(baseDir, file);
+                InputStream input = compileJsonnet(filePath.toString());
+                // Calling YamlConfigParser's parseStream method (instead of the overridden one below)
+                super.parseStream(collection, input, file);
+            } catch (NullPointerException e) {
+                collection.addError("File matching GoCD Jsonnet pattern disappeared", file);
+            }
+        }
+        return collection;   
+    }
+
+    @Override
+    public void parseStream(JsonConfigCollection result, InputStream input, String location) {
+        try {
+            InputStream jsonInputStream = compileJsonnet(input);
+            super.parseStream(result, jsonInputStream, location);
+        } catch (NullPointerException ex) {
+            result.addError("File matching GoCD Jsonnet pattern disappeared", location);
+        }
+    }
+
+    private InputStream compileJsonnet(String filePath) {
+        InputStream jsonInputStream = null;
+        try {
+            ProcessBuilder pb = new ProcessBuilder("jrsonnet", filePath, "--format", "yaml");
+            Process p = pb.start();
+            int exitCode = p.waitFor();
+            if (exitCode != 0) {
+                System.out.println("Error compiling jsonnet file: " + filePath);
+            }
+            jsonInputStream = p.getInputStream();
+        } catch (IOException e) {
+            System.out.println("Error compiling jsonnet file: " + e.getMessage());
+        } catch (InterruptedException e) {
+            System.out.println("Error compiling jsonnet file: " + e.getMessage());
+        }
```

Comment:
This seems to calmly continue when jrsonnet has failed.
That needs to be an error condition; there's no rational return value for this function if jrsonnet crashed.

```suggestion
            if (exitCode == 0) {
                return p.getInputStream();
            } else {
                throw Exception("jsonnet exited with an error");
            }
        } catch (Exception e) {
            throw JsonnetEvalException(("Error while evaluating jsonnet: " + e.getMessage());
        }
```